CRM/Contribute/BAO/Contribution.php

```diff
@@ -2883,8 +2883,8 @@ public static function recordFinancialAccounts(&$params, CRM_Contribute_DAO_Cont
     // is an entrenched mistake (the previous code did the same although
     // less obviously as it checked a partially populated contribution object.
     $isPayLater = !empty($params['is_pay_later']);
-    if (!$financialProcessor->isFailedTransaction() &&
-      !($financialProcessor->isPendingTransaction() && !$isPayLater)
+    $isIncompletePending = $financialProcessor->isPendingTransaction() && !$isPayLater;
+    if (!$financialProcessor->isFailedTransaction() && !$isIncompletePending
     ) {
```

Comment:
It's a little weird style-wise having the bracket here, but otherwise this PR looks ok.

Reply:
I fixed it @demeritcowboy